examples/js/cameras/CinematicCamera.js

```diff
@@ -0,0 +1,184 @@
+THREE.CinematicCamera.prototype.setLens = function ( focalLength, frameHeight, fNumber, coc ) {
+
+  //In case of cinematicCamera, having a default lens set is important
+  if ( focalLength === undefined ) focalLength = 35;
+
+  THREE.PerspectiveCamera.prototype.setLens.call( this, focalLength, frameHeight );
+
+  //if fnumber and coc are not provided, cinematicCamera tries to act as a basic PerspectiveCamera
+  if ( fNumber === undefined ) fNumber = 8;
+  if ( coc === undefined ) coc = 0.019;
+
+
+
+  this.focalLength = focalLength;
+  this.fNumber = fNumber;
+  this.coc = coc;
+
+  //fNumber is focalLength by aperture
+  this.aperture = this.focalLength/this.fNumber;
+
+  //hyperFocal is required to calculate depthOfField when a lens tries to focus at a distance with given fNumber and focalLength
+  this.hyperFocal = (this.focalLength*this.focalLength)/(this.aperture*this.coc);
+
+};
+
+
+
+THREE.CinematicCamera.prototype.linearize = function(depth) {
+	var zfar = this.far;
+	var znear = this.near;
+	return -zfar * znear / (depth * (zfar - znear) - zfar);
+}
+
+
+THREE.CinematicCamera.prototype.smoothstep = function(near, far, depth) {
+	var x = this.saturate( (depth - near) / (far - near));
+	return x * x * (3- 2*x);
+}
+
+
+THREE.CinematicCamera.prototype.saturate = function(x) {
+	return Math.max(0, Math.min(1, x));
+}
+
+
+//function for focusing at a distance from the camera
+THREE.CinematicCamera.prototype.focusAt = function (focusDistance){
+
+  if ( focusDistance === undefined ) focusDistance = 20;
+
+  //distance from the camera (normal to frustrum) to focus on (unused)
+  this.focusDistance = focusDistance;
+
+  //the nearest point from the camera which is in focus (unused)
+  this.nearPoint = (this.hyperFocal*this.focusDistance)/(this.hyperFocal+(this.focusDistance-this.focalLength));
+
+  //the farthest point from the camera which is in focus (unused)
+  this.farPoint = (this.hyperFocal*this.focusDistance)/(this.hyperFocal-(this.focusDistance-this.focalLength));
+
+  //the gap or width of the space in which is everything is in focus (unused)
+  this.depthOfField = this.farPoint - this.nearPoint;
+
+  //Considering minimum distance of focus for a standard lens (unused)
+  if( this.depthOfField < 0) this.depthOfField = 0;
+  
+  this.sdistance = this.smoothstep(this.near, this.far, this.focusDistance);
+
+	this.ldistance = this.linearize(1 -  this.sdistance);
+	
+	this.postprocessing.bokeh_uniforms[ 'focalDepth' ].value = this.ldistance;
+
+};
+
+
+
+THREE.CinematicCamera.prototype.initPostProcessing = function (){
+    if(this.postprocessing.enabled)
+    {
+        this.postprocessing.scene = new THREE.Scene();
+
+        
+        this.postprocessing.camera = new THREE.OrthographicCamera( window.innerWidth / -2, window.innerWidth / 2,  window.innerHeight / 2, window.innerHeight / - 2, -10000, 10000 );
+
+        this.postprocessing.camera.position= this.position;
+				this.postprocessing.camera.rotation= this.rotation;
```

Comment:
I wanted to set the position just to check, but dint needed it later. will change...